src/main/java/fr/esgi/al5_2/Tayarim/services/ProprietaireService.java

```diff
@@ -16,141 +16,172 @@
 import org.springframework.transaction.annotation.Transactional;
 import org.springframework.util.ObjectUtils;
 
+import java.security.SecureRandom;
 import java.util.List;
 import java.util.Optional;
 
 @Service
 @Transactional(readOnly = true)
 public class ProprietaireService {
 
-    private final ProprietaireRepository proprietaireRepository;
+  private final ProprietaireRepository proprietaireRepository;
 
-    public ProprietaireService(ProprietaireRepository proprietaireRepository) {
-        this.proprietaireRepository = proprietaireRepository;
+  public ProprietaireService(ProprietaireRepository proprietaireRepository) {
+    this.proprietaireRepository = proprietaireRepository;
+  }
+
+  @Transactional
+  public ProprietaireDTO creerProprietaire(
+      @NonNull ProprietaireCreationDTO proprietaireCreationDTO) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.naming.AbbreviationAsWordInNameCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Abbreviation in name 'proprietaireCreationDTO' must contain no more than '1' consecutive capital letters.